src/sql/workbench/parts/query/browser/queryActions.ts

```diff
@@ -415,6 +416,65 @@ export class ListDatabasesAction extends QueryTaskbarAction {
 	}
 }
 
+/**
+ * Action class that toggles SQLCMD mode for the editor
+ */
+export class ToggleSqlCmdModeAction extends QueryTaskbarAction {
+
+	public static EnableSqlcmdClass = 'enablesqlcmd';
+	public static DisableSqlcmdClass = 'disablesqlcmd';
+	public static ID = 'ToggleSqlCmdModeAction';
+
+	private _enablesqlcmdLabel = nls.localize('enablesqlcmdLabel', "Enable SQLCMD Mode");
+	private _disablesqlcmdLabel = nls.localize('disablesqlcmdLabel', "Disable SQLCMD Mode");
+	constructor(
+		editor: QueryEditor,
+		private _isSqlCmdMode: boolean,
+		@IQueryManagementService protected readonly queryManagementService: IQueryManagementService,
+		@IConfigurationService protected readonly configurationService: IConfigurationService,
+		@IConnectionManagementService connectionManagementService: IConnectionManagementService
+	) {
+		super(connectionManagementService, editor, ToggleSqlCmdModeAction.ID, undefined);
+	}
+
+	public get isSqlCmdMode(): boolean {
+		return this._isSqlCmdMode;
+	}
+
+	public set isSqlCmdMode(value: boolean) {
+		this._isSqlCmdMode = value;
+		this.updateLabelAndIcon();
+	}
+
+	private updateLabelAndIcon(): void {
+		if (this.isSqlCmdMode) {
+			// show option to disable sql cmd mode if already enabled
+			this.label = this._disablesqlcmdLabel;
+			this.updateCssClass(ToggleSqlCmdModeAction.DisableSqlcmdClass);
+		} else {
+			this.label = this._enablesqlcmdLabel;
+			this.updateCssClass(ToggleSqlCmdModeAction.EnableSqlcmdClass);
+		}
+	}
+
+	public run(): Promise<void> {
+		let queryoptions: QueryExecutionOptions = { options: new Map<string, any>() };
+		if (this.isSqlCmdMode) {
```

Comment:
same here, can be simplified to 4 lines

Reply:
Sorry for late reply. So for this one - the this.editor.input.state.isSqlCmdMode change triggers events that can change this.isSqlCMdMode. that's why wanted to keep it all together inside one if {}. but I will change to string the value in local var and using that in later statement. Let me know if that sounds okay.